test/test_field.jl

```diff
@@ -451,4 +450,54 @@ end
             end
         end
     end
+
+    @testset "Views of field views" begin
+        @info "  Testing views of field views..."
+
+        Nx, Ny, Nz = 1, 1, 7
+
+        FieldTypes = (CenterField, XFaceField, YFaceField, ZFaceField)
+        ZTopologies = (Periodic, Bounded)
+
+        for arch in archs, FT in float_types, FieldType in FieldTypes, ZTopology in ZTopologies
+            grid = RectilinearGrid(arch, FT, size=(Nx, Ny, Nz), x=(0, 1), y=(0, 1), z=(0, 1), topology = (Periodic, Periodic, ZTopology))
+            Hx, Hy, Hz = halo_size(grid)
+
+            k_top = (FieldType == ZFaceField && ZTopology == Bounded) ? Nz+1 : Nz 

```

Comment:
there is a function for this; you need to do `using Oceananigans.Grids: total_length` and then 

```suggestion
            k_top = total_length(location(c, 3)(), topology(c, 3)(), size(grid, 3))
```